Regenerate the sample Kubernetes manifests

L3n41c opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?

Regenerate the sample kubernetes manifests in `Dockerfiles/manifests` from the latest version of the Helm chart.

### Motivation

They were out-of-date.

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

L3n41c committed:
Regenerate the sample Kubernetes manifests

clamoriniere commented on Dockerfiles/manifests/agent-only/README.md:
```diff
@@ -1,11 +1,13 @@
   collectEvents: true
   leaderElection: true
+  apm:
+    socketEnabled: false
```
so we want do disable APM/trace-agent completely ?

L3n41c replied:
I only wanted to have the same things in the generated manifests.
We used to not have `trace-agent` in the manifests generated from this `agent-only` `values.yaml` because it was disabled by default.
But the Helm chart has been updated and the `trace-agent` is now enabled by default in the Helm chart.
But as far as the statically generated manifests are concerned, I thought it makes more sense to still not have the `trace-agent` in the `agent-only` manifests and to have it only in the `all-containers` manifests like we used to have.

clamoriniere commented on Dockerfiles/manifests/agent-only/agent-services.yaml:
```diff
@@ -13,3 +14,24 @@ spec:
+      port: 8125
+      targetPort: 8125
+      name: dogstatsd
+  internalTrafficPolicy: Local
```
I think it will be compatible only with k8s 1.22

L3n41c replied:
I’ve juste added `--kube-version 1.21` to `helm template` to skip the generation of objects that might not be supported on too old versions of k8s: https://github.com/DataDog/datadog-agent/pull/10393/commits/a772265694ea77c45a8aa01bc669833f291033c4.

clamoriniere commented on Dockerfiles/manifests/agent-only/daemonset.yaml:
```diff
@@ -185,10 +185,27 @@ spec:
             path: /sys/fs/cgroup
           name: cgroups
-        - emptyDir: {}
+        - hostPath:
+            path: /etc/os-release
+          name: etc-os-release
+        - hostPath:
+            path: /etc/redhat-release
+          name: etc-redhat-release
+        - hostPath:
+            path: /etc/fedora-release
+          name: etc-fedora-release
+        - hostPath:
+            path: /etc/lsb-release
+          name: etc-lsb-release
+        - hostPath:
+            path: /var/run/datadog/
+            type: DirectoryOrCreate
```
maybe a bug in our chart, but this should be mounted only if system-probe is.

L3n41c committed:
Regenerate the manifests with `--kube-version` 1.21

to skip the generation of the internal traffic policy service that might not be supported on targeted systems.

L3n41c commented on Dockerfiles/manifests/agent-only/daemonset.yaml:
```diff
@@ -185,10 +185,27 @@ spec:
             path: /sys/fs/cgroup
           name: cgroups
-        - emptyDir: {}
+        - hostPath:
+            path: /etc/os-release
+          name: etc-os-release
+        - hostPath:
+            path: /etc/redhat-release
+          name: etc-redhat-release
+        - hostPath:
+            path: /etc/fedora-release
+          name: etc-fedora-release
+        - hostPath:
+            path: /etc/lsb-release
+          name: etc-lsb-release
```
Good catch !
Should be addressed by DataDog/helm-charts#509.

L3n41c replied:
Now fixed in https://github.com/DataDog/datadog-agent/pull/10393/commits/3acdbfa46aa4c2a0c1c3f8d3680dbf5f4edcc9e9.

L3n41c committed:
Regenerate to take DataDog/helm-charts#509 into account

L3n41c committed:
Merge branch 'main' of github.com:DataDog/datadog-agent into lenaic/regenerate-k8s-manifests
